src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/HaFlowPingSpec.groovy

```diff
@@ -0,0 +1,212 @@
+import org.openkilda.northbound.dto.v2.haflows.HaFlowPingPayload
+import org.openkilda.northbound.dto.v2.haflows.HaFlowPingResult
+import org.openkilda.northbound.dto.v2.yflows.SubFlowPingPayload
+import org.openkilda.northbound.dto.v2.yflows.UniSubFlowPingPayload
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import spock.lang.Ignore
+import spock.lang.Narrative
+import spock.lang.Shared
+
+@Narrative("""This spec tests 'periodic ping' functionality.""")
+class HaFlowPingSpec extends HealthCheckSpecification {
+    @Autowired
+    @Shared
+    HaFlowHelper haFlowHelper
+
+    @Value('${flow.ping.interval}')
+    int pingInterval
+
+
+    @Tidy
+    @Tags([LOW_PRIORITY])
+    def "Able to turn off periodic pings on a HA-flow"() {
+        given: "An HA-flow with periodic pings turned on"
+        def swT = topologyHelper.switchTriplets.find { SwitchTriplet.ALL_ENDPOINTS_DIFFERENT(it) }
+        def haFlowRequest = haFlowHelper.randomHaFlow(swT).tap {
+            it.periodicPings = true
+        }
+        def haFlow = haFlowHelper.addHaFlow(haFlowRequest)
+        assert northboundV2.getHaFlow(haFlow.haFlowId).periodicPings
+
+        when: "Turn off periodic pings"
+        def updatedHaFlow = haFlowHelper.partialUpdateHaFlow(
+                haFlow.haFlowId, HaFlowPatchPayload.builder().periodicPings(false).build())
+
+        then: "Periodic pings are really disabled"
+        !updatedHaFlow.periodicPings
+        !northboundV2.getHaFlow(haFlow.haFlowId).periodicPings
+
+        and: "Packet counter on catch ping rules doesn't grow"
+        pingRuleCountersDoesntGrow(swT, haFlow)
+
+        cleanup:
+        haFlow && haFlowHelper.deleteHaFlow(haFlow.haFlowId)
+    }
+
+    @Tidy
+    @Tags([LOW_PRIORITY])
+    def "Unable to ping HA-flow via periodic pings if ISL is broken"() {
+        given: "Pinned HA-flow with periodic pings turned on"
+        // We need to use pinned flow because it wouldn't be rerouted from failed ISL
+        def swT = topologyHelper.switchTriplets.find { SwitchTriplet.ALL_ENDPOINTS_DIFFERENT(it) }
+        def haFlowRequest = haFlowHelper.randomHaFlow(swT).tap {
+            it.periodicPings = true
+            it.pinned = true
+        }
+        def haFlow = haFlowHelper.addHaFlow(haFlowRequest)
+        assert northboundV2.getHaFlow(haFlow.haFlowId).periodicPings
+        def paths = northboundV2.getHaFlowPaths(haFlow.haFlowId)
+        def islToFail = pathHelper.getInvolvedIsls(PathHelper.convert(paths.subFlowPaths[0].forward)).first()
+
+        when: "Fail an HA-flow ISL (bring switch port down)"
+        antiflap.portDown(islToFail.srcSwitch.dpId, islToFail.srcPort)
+        wait(WAIT_OFFSET) { northbound.getLink(islToFail).state == FAILED }
+
+        then: "Periodic pings are still enabled"
+        northboundV2.getHaFlow(haFlow.haFlowId).periodicPings
+
+        and: "Packet counter on catch ping rules doesn't grow"
```

Comment:
Don't we have some better 'outcome' of failed periodic ping than packet counter? Is the global purpose of periodic ping is statistics only?